webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/healthcheck/HealthcheckService.scala

```diff
@@ -0,0 +1,58 @@
+package com.normation.rudder.services.healthcheck
+
+import java.lang.Runtime.getRuntime
+
+import com.normation.errors.IOResult
+import com.normation.rudder.domain.logger.{HealthcheckLoggerPure => logger}
+import zio.UIO
+import zio.ZIO
+
+sealed trait Status
+case object Ok extends Status
+case object Warning extends Status
+case object Critical extends Status
+
+trait Check {
+  def name: String
+  def run: IOResult[HealthcheckResult]
+}
+
+final case class HealthcheckResult(
+    name  : String
+  , msg   : String
+  , status: Status
+)
+
+class HealthcheckService(checks: List[Check]) {
+
+   def runAll: ZIO[Any, Nothing, List[HealthcheckResult]]  = ZIO.foreach(checks) { c =>
+      for {
+        res <- c.run.catchAll { err =>
+                 HealthcheckResult(c.name, s"A fatal error was encountered when running check '${c.name}': ${err.fullMsg}", Critical)
```

Comment:
I get a `RudderError => HealthcheckResult` instead of `RudderError => ZIO[R,E,A] `

Reply:
look at catchAll sig, maybe you need a `.succeed`, or maybe it's not catchAll but an other one that allows to change an error to a success. You need something with signature `E => ZIO[R, E1, B]`

Reply:
You need to encapsulare your HealthCheckResult in a zio 

just call succeed function at the end

```
                 HealthcheckResult(c.name, s"A fatal error was encountered when running check '${c.name}': ${err.fullMsg}", Critical).succeed
```